source/slang/slang-emit-c-like.h

```diff
@@ -463,7 +471,7 @@ class CLikeSourceEmitter: public SourceEmitterBase
 
     virtual void emitPostDeclarationAttributesForType(IRInst* type) { SLANG_UNUSED(type); }
     virtual bool doesTargetSupportPtrTypes() { return false; }
-    virtual void emitLayoutSemanticsImpl(IRInst* inst, char const* uniformSemanticSpelling = "register") { SLANG_UNUSED(inst); SLANG_UNUSED(uniformSemanticSpelling); }
+    virtual void emitLayoutSemanticsImpl(IRInst* inst, char const* uniformSemanticSpelling = "register", EmitLayoutSemanticOption layoutSemanticOption = EmitLayoutSemanticOption::kPostType) { SLANG_UNUSED(inst); SLANG_UNUSED(uniformSemanticSpelling); SLANG_UNUSED(layoutSemanticOption); }
```

Comment:
Can we avoid using the default value especially when the function is a virtual function.
Because the behavior can be often unexpected and I have seen a few hard bugs stemming from this.

Reply:
Should I also change the old behavior of `char const* uniformSemanticSpelling = "register"` to no longer use the "register" default as well?

Reply:
I am not sure what "behavior" you are referring to.
I assume you can remove `= "register"` and change the rest of code to make them working.